code/modules/food_and_drinks/kitchen_machinery/deep_fryer.dm

```diff
@@ -0,0 +1,315 @@
++######################+++++++++++++++++++++++++++++++++++'''''''+++++++++++++++#################################'''''''''''''+''+''';;'''';;;;;;;''';:,:,::::;;::::::::;;;:;;;:;;';;;;;;;;;;;;;;:,,,,,.
++#######################++++++++++++++++++++++++++++++++++''''+''++++++++++++++++#################################+''''''''''''''''''';;''';;;;;;;'''';:::::::;;:,::::::;;;;;;;;;;;;;;;;;;;;;;;;;:,,,,,,
++########################+++++++++++++++++++++++++++++++''''''++++''++++++++++++####################################+'''''''''''''''''';'''';;;;;'''''';::::::::;::::::::;;;;;:;;;;;;;;;;;;;;;;;;::,,,,,
+++########################+++++++++++++++++++++++++++++++''''+'+++''+++++++++++++#####################################'''''''''''''''''';''';;;;;;'''''';;;;;::::::::::::::::;;;;;;;;;;;;;;;;;;;;::,,,,,
++++######################+#++++++++++++++++++++++++++++''''''++++++'++++++++++++++######################################+''''''''''''''+;;'''';;'''''''''';'';;::::::::::::::;;;;;;;;;;;;;;;;;;;:::,,,,,
+##########################+++++++++++++++++++++++++++++''''''++++++'+++++++++++++#########################################+'''''''''''''';''''+;;''''''''''';'';::::::::::::::;;;;;;;';;;;;;;;;;:::,,,,,
+##########################+++++++++++++++++++++++++++++'''''+++++'''++++++++++++++##########################################+''''''''''''';+'''';;''''''''''''';';;;::::::::::;;;';;;;;;';;;;;;;;:::,,,,
+++#######################+##+++++++++++++++++++++++++++'''''++++++''++++++++++++++#########################################++''''''''''''+':'+''';'''''''';'';'';;;:::::::::::;;;';;;;;;;;;;;;;;;:::,,,,
+*/
+
+// April 3rd, 2014 marks the day this machine changed the face of the kitchen on NTStation13
+// God bless America.
+/obj/machinery/deepfryer
+	name = "deep fryer"
+	desc = "Deep fried <i>everything</i>."
+	icon = 'icons/obj/kitchen.dmi'
+	icon_state = "fryer_off"
+	density = 1
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 5
+	container_type = OPENCONTAINER
+	var/obj/item/frying = null	//What's being fried RIGHT NOW?
+	var/cook_time = 0
+
+/obj/item/weapon/circuitboard/machine/deep_fryer
+	name = "circuit board (Deep Fryer)"
+	build_path = /obj/machinery/deepfryer
+	origin_tech = "programming=1"
+	req_components = list(/obj/item/weapon/stock_parts/micro_laser = 1)
+
+/obj/machinery/deepfryer/New()
+	..()
+	create_reagents(50)
+	reagents.add_reagent("nutriment", 25)
+	component_parts = list()
+	component_parts += new /obj/item/weapon/circuitboard/machine/deep_fryer(null)
+	component_parts += new /obj/item/weapon/stock_parts/micro_laser(null)
+	RefreshParts()
+
+/obj/machinery/deepfryer/examine()
+	..()
+	if(frying)
+		usr << "You can make out [frying] in the oil."
+
+/obj/machinery/deepfryer/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/reagent_containers/food/snacks/deepfryholder))
+		user << "<span class='userdanger'>Your cooking skills are not up to the legendary Doublefry technique.</span>"
```

Comment:
@ma44 it needs to be a var in mind and you can set it when you equip the chefs  as well as with other objects